derive-impl/src/pyclass.rs

```diff
@@ -413,8 +413,65 @@ pub(crate) fn impl_pyclass(attr: AttributeArgs, item: Item) -> Result<TokenStrea
         attrs,
     )?;
 
+    const ALLOWED_TRAVERSE_OPTS: &[&str] = &["manual"];
+    // try to know if it have a `#[pyclass(trace)]` exist on this struct
+    // TODO(discord9): rethink on auto detect `#[Derive(PyTrace)]`
+    let trace_opt = {
+        class_meta
+            .inner()
+            ._optional_key_with_optional_str("traverse")?
+    };
+    let (maybe_trace_code, derive_trace) = match trace_opt {
+        Some(Some(s)) => {
+            if !ALLOWED_TRAVERSE_OPTS.contains(&s.as_str()) {
+                bail_span!(
+                    item,
+                    "trace attribute only accept {ALLOWED_TRAVERSE_OPTS:?} as value or no value at all",
+                );
+            }
+            (
+                quote! {
+                    impl ::rustpython_vm::object::gc::MaybeTraverse for #ident {
+                        const IS_TRACE: bool = true;
+                        fn try_traverse(&self, tracer_fn: &mut ::rustpython_vm::object::gc::TraverseFn) {
+                            ::rustpython_vm::object::gc::Traverse::traverse(self, tracer_fn);
+                        }
+                    }
+                },
+                quote! {},
+            )
+        }
+        Some(None) => (
+            quote! {
+                impl ::rustpython_vm::object::gc::MaybeTraverse for #ident {
+                    const IS_TRACE: bool = true;
+                    fn try_traverse(&self, tracer_fn: &mut ::rustpython_vm::object::gc::TraverseFn) {
+                        ::rustpython_vm::object::gc::Traverse::traverse(self, tracer_fn);
+                    }
+                }
+            },
+            quote! {#[derive(PyTraverse)]},
+        ),
```

Comment:
```suggestion
        Some(value) => {
            let derive_trace = match value {
                Some(s) => {
                    if !ALLOWED_TRAVERSE_OPTS.contains(&s.as_str()) {
                        bail_span!(
                            item,
                            "trace attribute only accept {ALLOWED_TRAVERSE_OPTS:?} as value or no value at all",
                        );
                    }
                    quote! {}
                }
                None => quote! {#[derive(PyTraverse)]},
            };
            (
                quote! {
                    impl ::rustpython_vm::object::gc::MaybeTraverse for #ident {}
                },
                derive_trace
            )
        }
```

can MaybeTraverse have traverse as its default implementation? It will reduce generated code footprint

Reply:
@youknowone would something like this work (rewriting the inner `match` to be an `if let ... else`)?

```rust
Some(value) => {
    let derive_trace = if let Some(s) = value {
        if !ALLOWED_TRAVERSE_OPTS.contains(&s.as_str()) {
            bail_span!(
                item,
                "trace attribute only accept {ALLOWED_TRAVERSE_OPTS:?} as value or no value at all",
            );
        }
        quote! {}
    } else {
        quote! {#[derive(PyTraverse)]}
    };
}
```

Reply:
sure, that looks better.

Reply:
> can MaybeTraverse have traverse as its default implementation? It will reduce generated code footprint

That might be not possible until negative impl or optional impl is available? Like T: Traverse for MaybeTraverse & T: !Traverse for MaybeTrace?